src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/ListExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+    internal static class ListExtensions
+    {
+        public static IList<T> Shuffle<T>(this IList<T> values)
+        {
+            var rdm = new Random();
+            int count = values.Count;

```

Comment:
Since we're checking for null on other public methods, seems like we should do the same for `values` here.

Reply:
I agree most cases but extension methods are a bit special, and it's based off the usage. The consumption of this method would be like

`myList.Shuffle();`

Extension methods are really supposed to act like they are methods of the object they are extending. Given such a design pattern, a null-ref would be expected if `myList` is null in this case. Thus, we don't check for null on the `this` parameter in extension methods.